boards/arduino-uno/include/periph_conf.h

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright (C) 2014 Freie Universität Berlin, Hinnerk van Bruinehsen
+ *               2016 Laurent Navet <laurent.navet@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     boards_arduino-uno
+ * @{
+ *
+ * @file
+ * @brief       Peripheral MCU configuration for the Arduino Uno board
+ *
+ * @author      Hinnerk van Bruinehsen <h.v.bruinehsen@fu-berlin.de>
+ * @author      Laurent Navet <laurent.navet@gmail.com>
+ */
+
+#ifndef PERIPH_CONF_H_
+#define PERIPH_CONF_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Clock configuration
+ * @{
+ */
+#define CLOCK_CORECLOCK     (16000000L)
+/** @} */
+
+/**
+ * @brief   Timer configuration
+ *
+ * The timer driver only supports the 16-bit timer (Timer1)
+ * so this is the only one we can use here.
+ *
+ * @{
+ */
+#define TIMER_NUMOF         (2U)
+
+#define TIMER_0             MEGA_TIMER1
+#define TIMER_0_MASK        &TIMSK1
+#define TIMER_0_FLAG        &TIFR1
+#define TIMER_0_ISRA        TIMER1_COMPA_vect
+#define TIMER_0_ISRB        TIMER1_COMPB_vect
+/** @} */
+
+/**
+ * @brief   UART configuration
+ * @{
+ */
+#define UART_NUMOF          (1U)
+#define UART_0_EN           1
+
+/* UART 0 registers */
+#define UART0_CTRL_STAT_A   UCSR0A
+#define UART0_CTRL_STAT_B   UCSR0B
+#define UART0_CTRL_STAT_C   UCSR0C
+#define UART0_BAUD_RATE_L   UBRR0L
+#define UART0_BAUD_RATE_H   UBRR0H
+#define UART0_DATA_REGISTER UDR0
+
+/* Flags */
+#define UART0_RX_COMPLETE   RXC0
+#define UART0_DATA_EMPTY    UDRE0
+#define UART0_RX_EN         RXEN0
+#define UART0_TX_EN         TXEN0
+#define UART0_RXC_IRQ_EN    RXCIE0
+#define UART0_TXC_IRQ_EN    TXCIE0
+#define UART0_8BIT_SIZE     (1 << UCSZ00) | (1 << UCSZ01)
+
+/* UART0 helper macros */
+#define UART0_RX_TX_EN      UART0_CTRL_STAT_B |= (1 << UART0_RX_EN) | (1 << UART0_TX_EN)
+#define UART0_RX_IRQ_EN     UART0_CTRL_STAT_B |= (1 << UART0_RXC_IRQ_EN)
+#define UART0_SET_8BIT_SIZE UART0_CTRL_STAT_C |= UART0_8BIT_SIZE
+#define UART0_RECEIVED_DATA (UART0_CTRL_STAT_A & (1 << UART0_RX_COMPLETE))
+#define UART0_DTREG_EMPTY   (UART0_CTRL_STAT_A & (1 << UART0_DATA_EMPTY))
+
```

Comment:
Please look at the new UART implementation, for instance on the mega2560 [here](https://github.com/RIOT-OS/RIOT/blob/master/boards/arduino-mega2560/include/periph_conf.h#L69).